artemis/modules/sql_injection_detector.py

```diff
@@ -0,0 +1,187 @@
+import datetime
+import random
+import re
+import urllib
+from timeit import default_timer as timer
+from typing import Any, Dict, List, Optional
+from urllib.parse import parse_qs, unquote, urlencode, urlparse, urlunparse
+
+import more_itertools
+from karton.core import Task
+
+from artemis import http_requests
+from artemis.binds import Service, TaskStatus, TaskType
+from artemis.config import Config
+from artemis.crawling import get_links_and_resources_on_same_domain
+from artemis.http_requests import HTTPResponse
+from artemis.karton_utils import check_connection_to_base_url_and_save_error
+from artemis.module_base import ArtemisBase
+from artemis.sql_messages_example import SQL_ERROR_MESSAGES, URL_PARAMS
+from artemis.task_utils import get_target_url
+
+
+class SqlInjectionDetector(ArtemisBase):
+    """
+    Module for detecting sql Injection and Base Time Sql Injection vulnerabilities.
+    """
+
+    identity = "sql_injection_detector"
+    filters = [
+        {"type": TaskType.SERVICE.value, "service": Service.HTTP.value},
+    ]
+
+    @staticmethod
+    def _strip_query_string(url: str) -> str:
+        url_parsed = urllib.parse.urlparse(url)
+        return urllib.parse.urlunparse(url_parsed._replace(query="", fragment=""))
+
+    @staticmethod
+    def create_url_with_batch_payload(
+        url: str, param_batch: Optional[tuple[Any]], sleep_payload: Optional[str] = None
+    ) -> str:
+        if param_batch is not None:
+            assignments = {
+                key: sleep_payload if sleep_payload else random.choice(Config.Modules.SqlInjectionDetector.PAYLOADS)
```

Comment:
I think a better idea would be to always expect the payload and the caller should check all four:

- `'`
- `"`
- `sleep`
- `pg_sleep`

Reply:
Done.